src/createDeeplink.ts

```diff
@@ -0,0 +1,118 @@
+import { environment } from "@raycast/api";
+
+export type DeeplinkType = "extension" | "script-command"
+
+export type CreateScriptCommandDeeplinkOptions = {
+  /**
+   * The type of deeplink, which should be "script-command".
+   */
+  type: "script-command",
+  /**
+   * The name of the command.
+   */
+  command: string,
+  /**
+   * If the command accepts arguments, they can be passed using this query parameter.
+   */
+  arguments?: string[],
+};
+
+/**
+ * Options for creating a deeplink to an extension.
+ */
+export type CreateExtensionDeeplinkOptions = {
+  /**
+   * The type of deeplink, which should be "extension".
+   */
+  type: "extension",
+  /**
+   * The name of the owner or author of the extension.
+   */
+  ownerOrAuthorName: string,
+  /**
+   * The name of the extension.
+   */
+  extensionName: string,
+  /**
+   * The command associated with the extension.
+   */
+  command: string,
+  /**
+   * Either "userInitiated", which runs the command in the foreground, or "background", which skips bringing Raycast to the front.
+   */
+  launchType?: "userInitiated" | "background",
+  /**
+   * If the command accepts arguments, they can be passed using this query parameter.
+   */
+  arguments?: {
+    /**
+     * The representation of arguments given that key here is the `name` defined in manifest file and value is the user's input
+     */
+    [item: string]: any;
+  },
+  /**
+   * If the command make use of LaunchContext, it can be passed using this query parameter.
+   */
+  context?: {
+    /**
+     * The context values for a command launch.
+     */
+    [item: string]: any;
+  },
```

Comment:
These type definitions are copied over from `@raycast/api` type definitions. Not sure if this is the best way to do this, or if we could somehow "import" those type definitions over here to prevent duplication

Reply:
Switched to using those types.

Reply:
Sadly `LaunchOptions` are not exported, so copy it seems to be the only way for now.

```tsx
 /**
  * Options for launching a command from the same extension or from another extension.
  */
 declare type LaunchOptions = IntraExtensionLaunchOptions | InterExtensionLaunchOptions;
```

@mathieudutour Maybe export it in the next version? I did the copy in [RaycastUI](https://github.com/tonka3000/raycastui/blob/0edf2950da1c793a9a08e39c56a8ceee61c55637/src/lib/rctypes.ts#L42) as well